### PR TITLE
Plt 5848 get contract

### PR DESCRIPTION
**Pull request by jhbertra**
Changes in this PR:

- [x] Updated and unified query protocol so chain-sync, marlowe-sync, and marlowe-contract all use the same query protocol.
- [x] Added new contract query API to `marlowe-contract`
- [x] Added contract query sub-protocol to `marlowe-proxy`
- [x] Added contract store query - get contract by hash
- [x] Added Runtime CLI query commands

Pre-submit checklist:
- Branch
    - [x] Tests are provided (if possible)
    - [ ] [Test report is updated](https://github.com/input-output-hk/marlowe-cardano/blob/main/marlowe/test/test-report.md) (if relevant)
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [ ] Relevant tickets are mentioned in commit messages
    - [x] Formatting, PNG optimization, etc. are updated
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
        - Review required
        - [x] Substantial changes to code, test, or documentation
        - [ ] Change made to Marlowe validator (@bwbush and @palas must be included as reviewers)
        - Review not required
        - [ ] Minor changes to non-critical code, documentation, nix derivations, configuration files, or scripts
        - [ ] Formatting, spelling, grammar, or reorganization
    - [x] Reviewer requested

### CODE REVIEW

File: marlowe-integration/src/Test/Integration/Marlowe/Local.hs [hunk 1]
```diff
@@ -316,6 +315,7 @@ withLocalMarloweRuntime' MarloweRuntimeOptions{..} test = withRunInIO \runInIO -
     contractStore <- createContractStore ContractStoreOptions
       { contractStoreDirectory = resolveWorkspacePath workspace "contract-store"
       , contractStoreStagingDirectory = resolveWorkspacePath workspace "contract-staging-area"
+      , lockingSleepBetweenRetries = 100_000
```
Comment (bwbush) [inline marlowe-integration/src/Test/Integration/Marlowe/Local.hs:318]: Presumably this is microseconds. I wonder if `lockingMicrosecondsBetweenRetries` might be more expressive and self documenting?

File: marlowe-protocols/src/Network/Protocol/Query/Types.hs [hunk 2]
```diff
@@ -1,401 +1,386 @@
...
-
-class IsQuery query => QueryVariations query where
-  tags :: NonEmpty (SomeTag query)
-  queryVariations :: Tag query delimiter err results -> NonEmpty (query delimiter err results)
-  delimiterVariations :: Tag query delimiter err results -> [delimiter]
-  errVariations :: Tag query delimiter err results -> [err]
-  resultsVariations :: Tag query delimiter err results -> NonEmpty results
-
-instance ArbitraryQuery query => ArbitraryMessage (Query query) where
+class Request req => ArbitraryRequest req where
```
Comment (bwbush) [inline marlowe-protocols/src/Network/Protocol/Query/Types.hs:246]: Why isn't this located in a separate package for test generators. Are these arbitrary instances needed by the protocol?

**Review by bwbush — APPROVED** [review]
Will the format/organization of the file store for contracts always be forward compatible, or do we have to version it? For some users, it might be the only record they have of the continuations, so they'll want it forever usable in new versions of Marlowe Runtime.

**Comment by jhbertra** [comment]
> Will the format/organization of the file store for contracts always be forward compatible, or do we have to version it? For some users, it might be the only record they have of the continuations, so they'll want it forever usable in new versions of Marlowe Runtime.

Good question. As long as the `.contract` files are always forward-compatible, we should be fine. The `.closure` and `.adjacency` files are basically just precomputed data from the `.contract` files. I don't see a reason why the `.contract` files would ever need to change.

**Pull request merged**